Environment variable not being parsed correctly

### To Reproduce

1. use this as an .env variable: `DATABASE_URL=mongodb://redacted:redacted@redacted:27017/redacted?directConnection=true&replicaSet=rs0&authSource=admin`
2. deploy
3. get an error from prisma
```
#17 4.766 Error: P1013
#17 4.766
#17 4.766 The provided database string is invalid. An invalid argument was provided: connection string options is not a `key=value` pair: directConnectiontrue in database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.
#17 4.766
#17 4.800  ELIFECYCLE  Command failed with exit code 1.
```


### Current vs. Expected behavior

the app deploys sucessfully without errors

### Provide environment information

```bash
OS: ubuntu server 24.04
host: proxmox vm
resources: 6 cores w/ 14gb ram
```

### Which area(s) are affected? (Select all that apply)

Application

### Are you deploying the applications where Dokploy is installed or on a remote server?

Same server where Dokploy is installed

### Additional context

full deployment log
Initializing deployment
Cloning into '/etc/dokploy/applications/polarlearn-polarlearn-y41mka/code'...
remote: Enumerating objects: 492, done.
Cloned github.com/polarnl/PolarLearn.git to /etc/dokploy/applications/polarlearn-polarlearn-y41mka/code: ✅
Preparing Railpack build plan...
╭─────────────────╮
│ Railpack 0.0.64 │
╰─────────────────╯
↳ Detected Node
↳ Using pnpm package manager
Packages
──────────
pnpm  │  9.15.9   │  railpack default (9)
node  │  22.17.0  │  railpack default (22)
Steps
──────────
▸ install
$ pnpm install --frozen-lockfile --prefer-offline
▸ build
$ pnpm run build
Deploy
──────────
$ pnpm run start
✅ Railpack prepare completed.
Building with Railpack frontend...
#0 building with "builder-containerd" instance using docker-container driver
#1 [internal] load build definition from railpack-plan.json
#1 transferring dockerfile: 3.68kB done
#1 DONE 0.0s
#2 resolve image config for docker-image://ghcr.io/railwayapp/railpack-frontend:v0.0.64
#2 DONE 0.2s
#3 docker-image://ghcr.io/railwayapp/railpack-frontend:v0.0.64@sha256:7278141308bef8c97508eb2f4b4f9f6a738afd0371d2ccc70b57f4c958b0bc6d
#3 resolve ghcr.io/railwayapp/railpack-frontend:v0.0.64@sha256:7278141308bef8c97508eb2f4b4f9f6a738afd0371d2ccc70b57f4c958b0bc6d done
#3 CACHED
#4 load build definition from railpack-plan.json
#4 transferring dockerfile: 3.68kB done
#4 DONE 0.0s
#5 loading .
#5 DONE 0.0s
#6 docker-image://ghcr.io/railwayapp/railpack-builder:latest
#6 resolve ghcr.io/railwayapp/railpack-builder:latest
#6 resolve ghcr.io/railwayapp/railpack-builder:latest 0.2s done
#6 DONE 0.2s
#7 docker-image://ghcr.io/railwayapp/railpack-runtime:latest
#7 resolve ghcr.io/railwayapp/railpack-runtime:latest
#7 resolve ghcr.io/railwayapp/railpack-runtime:latest 0.2s done
#7 CACHED
#8 mkdir /etc/mise
#8 CACHED
#9 create mise config
#9 CACHED
#10 install mise packages: node, pnpm
#10 CACHED
#11 copy /mise/shims
#11 CACHED
#12 copy /mise/installs
#12 CACHED
#13 copy /usr/local/bin/mise
#13 CACHED
#14 copy /etc/mise/config.toml
#14 CACHED
#15 copy /root/.local/state/mise
#15 CACHED
#5 loading .
#5 transferring context: 10.76MB 0.1s
#5 transferring context: 26.79MB 0.2s done
#5 DONE 0.2s
#10 install mise packages: node, pnpm
#10 CACHED
#16 copy .
#16 DONE 0.2s
#17 pnpm install --frozen-lockfile --prefer-offline
#17 0.356 Lockfile is up to date, resolution step is skipped
#17 0.444 Progress: resolved 1, reused 0, downloaded 0, added 0
#17 0.591 Packages: +1104
#17 0.591 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
#17 1.444 Progress: resolved 1104, reused 1104, downloaded 0, added 175
#17 2.446 Progress: resolved 1104, reused 1104, downloaded 0, added 445
#17 3.448 Progress: resolved 1104, reused 1104, downloaded 0, added 1081
#17 3.644 Progress: resolved 1104, reused 1104, downloaded 0, added 1104, done
#17 4.107
#17 4.107 dependencies:
#17 4.107 + @aws-sdk/client-s3 3.842.0
#17 4.107 + @dnd-kit/core 6.3.1
#17 4.107 + @dnd-kit/sortable 10.0.0
#17 4.107 + @dnd-kit/utilities 3.2.2
#17 4.107 + @fortawesome/fontawesome-svg-core 6.7.2
#17 4.107 + @fortawesome/free-solid-svg-icons 6.7.2
#17 4.107 + @fortawesome/react-fontawesome 0.2.2
#17 4.107 + @hookform/resolvers 5.1.1
#17 4.107 + @next/third-parties 15.4.0-canary.51
#17 4.107 + @prisma/client 6.10.1
#17 4.107 + @radix-ui/react-checkbox 1.3.2
#17 4.107 + @radix-ui/react-dialog 1.1.14
#17 4.107 + @radix-ui/react-dropdown-menu 2.1.15
#17 4.107 + @radix-ui/react-label 2.1.7
#17 4.107 + @radix-ui/react-popover 1.1.14
#17 4.107 + @radix-ui/react-progress 1.1.7
#17 4.107 + @radix-ui/react-radio-group 1.3.7
#17 4.107 + @radix-ui/react-scroll-area 1.2.9
#17 4.107 + @radix-ui/react-select 2.2.5
#17 4.107 + @radix-ui/react-separator 1.1.7
#17 4.107 + @radix-ui/react-slot 1.2.3
#17 4.107 + @radix-ui/react-switch 1.2.5
#17 4.107 + @radix-ui/react-tooltip 1.2.7
#17 4.107 + @sentry/nextjs 9.33.0
#17 4.107 + @tailwindcss/postcss 4.1.11
#17 4.107 + @tailwindcss/typography 0.5.16
#17 4.107 + @types/d3 7.4.3
#17 4.107 + @types/disposable-email 0.2.2
#17 4.107 + @types/nodemailer 6.4.17
#17 4.107 + @types/ws 8.18.1
#17 4.107 + argon2 0.43.0
#17 4.107 + babel-plugin-react-compiler 19.1.0-rc.2
#17 4.107 + class-variance-authority 0.7.1
#17 4.107 + clsx 2.1.1
#17 4.107 + cmdk 1.1.1
#17 4.107 + d3 7.9.0
#17 4.107 + disposable-email 0.2.3
#17 4.107 + google-translate-api-x 10.7.2
#17 4.107 + import-in-the-middle 1.14.2
#17 4.107 + jdenticon 3.3.0
#17 4.107 + jose 6.0.11
#17 4.107 + lottie-react 2.4.1
#17 4.107 + lucide-react 0.524.0
#17 4.107 + motion 12.19.2
#17 4.107 + next 15.4.0-canary.23
#17 4.107 + next-pwa 5.6.0
#17 4.107 + next-view-transitions 0.3.4
#17 4.107 + next-ws 2.0.9
#17 4.107 + nextstepjs 2.1.1
#17 4.107 + nodemailer 7.0.3
#17 4.107 + react 19.1.0
#17 4.107 + react-dom 19.1.0
#17 4.107 + react-hook-form 7.59.0
#17 4.107 + react-infinite-scroll-component 6.1.0
#17 4.107 + react-loader-spinner 6.1.6
#17 4.107 + react-markdown 10.1.0
#17 4.107 + react-scroll-motion 0.3.5
#17 4.107 + react-toastify 11.0.5
#17 4.107 + rehype-highlight 7.0.2
#17 4.107 + rehype-sanitize 6.0.0
#17 4.107 + remark-gfm 4.0.1
#17 4.107 + require-in-the-middle 7.5.2
#17 4.107 + sharp 0.34.2
#17 4.107 + tailwind-merge 3.3.1
#17 4.107 + tailwindcss-animate 1.0.7
#17 4.107 + uuid 11.1.0
#17 4.107 + ws 8.18.3
#17 4.107 + zod 3.25.67
#17 4.107
#17 4.107 devDependencies:
#17 4.107 + @eslint/js 9.30.0
#17 4.107 + @types/node 24.0.7
#17 4.107 + @types/react 19.1.8
#17 4.107 + @types/react-dom 19.1.6
#17 4.107 + eslint 9.30.0
#17 4.107 + eslint-config-prettier 10.1.5
#17 4.107 + eslint-plugin-import 2.32.0
#17 4.107 + eslint-plugin-react 7.37.5
#17 4.107 + globals 16.2.0
#17 4.107 + postcss 8.5.6
#17 4.107 + prisma 6.10.1
#17 4.107 + tailwindcss 4.1.11
#17 4.107 + typescript 5.8.3
#17 4.107 + typescript-eslint 8.35.0
#17 4.107
#17 4.139
#17 4.139 > polarlearn@v1-PRERELEASE-1.1 postinstall /app
#17 4.139 > prisma db push
#17 4.139
#17 4.742 Prisma schema loaded from prisma/schema.prisma
#17 4.742 Datasource "db": MongoDB database "redacted" at "192.168.86.105:27017"
#17 4.766 Error: P1013
#17 4.766
#17 4.766 The provided database string is invalid. An invalid argument was provided: connection string options is not a `key=value` pair: directConnectiontrue in database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.
#17 4.766
#17 4.800  ELIFECYCLE  Command failed with exit code 1.
#17 ERROR: process "pnpm install --frozen-lockfile --prefer-offline" did not complete successfully: exit code: 1
------
> pnpm install --frozen-lockfile --prefer-offline:
4.139 > polarlearn@v1-PRERELEASE-1.1 postinstall /app
4.139 > prisma db push
4.139
4.742 Prisma schema loaded from prisma/schema.prisma
4.742 Datasource "db": MongoDB database "polarlearn" at "192.168.86.105:27017"
4.766 Error: P1013
4.766
4.766 The provided database string is invalid. An invalid argument was provided: connection string options is not a `key=value` pair: directConnectiontrue in database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.
4.766
4.800  ELIFECYCLE  Command failed with exit code 1.
------
ERROR: failed to build: failed to solve: process "pnpm install --frozen-lockfile --prefer-offline" did not complete successfully: exit code: 1
❌ Railpack build failed

### Will you send a PR to fix it?

Maybe, need help

With debug logs enabled:
Initializing deployment
Cloning into '/etc/dokploy/applications/polarlearn-polarlearn-y41mka/code'...
remote: Enumerating objects: 492, done.
Cloned github.com/polarnl/PolarLearn.git to /etc/dokploy/applications/polarlearn-polarlearn-y41mka/code: ✅
Preparing Railpack build plan...
╭─────────────────╮
│ Railpack 0.0.64 │
╰─────────────────╯
↳ Detected Node
↳ Using pnpm package manager
Packages
──────────
node  │  22.17.0  │  railpack default (22)
pnpm  │  9.15.9   │  railpack default (9)
Steps
──────────
▸ install
$ pnpm install --frozen-lockfile --prefer-offline
▸ build
$ pnpm run build
Deploy
──────────
$ pnpm run start
✅ Railpack prepare completed.
Building with Railpack frontend...
#0 building with "builder-containerd" instance using docker-container driver
#1 [internal] load build definition from railpack-plan.json
#1 transferring dockerfile: 3.68kB done
#1 DONE 0.0s
#2 resolve image config for docker-image://ghcr.io/railwayapp/railpack-frontend:v0.0.64
#2 DONE 0.4s
#3 docker-image://ghcr.io/railwayapp/railpack-frontend:v0.0.64@sha256:7278141308bef8c97508eb2f4b4f9f6a738afd0371d2ccc70b57f4c958b0bc6d
#3 resolve ghcr.io/railwayapp/railpack-frontend:v0.0.64@sha256:7278141308bef8c97508eb2f4b4f9f6a738afd0371d2ccc70b57f4c958b0bc6d 0.0s done
#3 CACHED
#4 load build definition from railpack-plan.json
#4 transferring dockerfile: 3.68kB done
#4 DONE 0.0s
#5 [railpack] secrets hash
#5 DONE 0.0s
#6 loading .
#6 DONE 0.0s
#7 docker-image://ghcr.io/railwayapp/railpack-builder:latest
#7 resolve ghcr.io/railwayapp/railpack-builder:latest
#7 resolve ghcr.io/railwayapp/railpack-builder:latest 0.3s done
#7 DONE 0.3s
#8 docker-image://ghcr.io/railwayapp/railpack-runtime:latest
#8 resolve ghcr.io/railwayapp/railpack-runtime:latest
#8 resolve ghcr.io/railwayapp/railpack-runtime:latest 0.3s done
#8 CACHED
#9 copy /etc/mise/config.toml
#9 CACHED
#10 create mise config
#10 CACHED
#11 copy /mise/shims
#11 CACHED
#12 copy /mise/installs
#12 CACHED
#13 copy /usr/local/bin/mise
#13 CACHED
#14 install mise packages: node, pnpm
#14 CACHED
#15 mkdir /etc/mise
#15 CACHED
#16 copy /root/.local/state/mise
#16 CACHED
#5 [railpack] secrets hash
#5 CACHED
#6 loading .
#6 transferring context: 11.03MB 0.1s
#6 transferring context: 26.79MB 0.2s done
#6 DONE 0.2s
#14 install mise packages: node, pnpm
#14 CACHED
#17 copy .
#17 DONE 0.2s
#18 pnpm install --frozen-lockfile --prefer-offline
#18 0.348 Lockfile is up to date, resolution step is skipped
#18 0.434 Progress: resolved 1, reused 0, downloaded 0, added 0
#18 0.609 Packages: +1104
#18 0.609 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
#18 1.433 Progress: resolved 1104, reused 0, downloaded 114, added 113
#18 2.433 Progress: resolved 1104, reused 0, downloaded 174, added 174
#18 3.433 Progress: resolved 1104, reused 0, downloaded 229, added 229
#18 4.433 Progress: resolved 1104, reused 0, downloaded 297, added 297
#18 5.433 Progress: resolved 1104, reused 0, downloaded 378, added 377
#18 6.433 Progress: resolved 1104, reused 0, downloaded 444, added 444
#18 7.433 Progress: resolved 1104, reused 0, downloaded 516, added 516
#18 8.435 Progress: resolved 1104, reused 0, downloaded 640, added 640
#18 9.434 Progress: resolved 1104, reused 0, downloaded 832, added 832
#18 10.43 Progress: resolved 1104, reused 0, downloaded 1008, added 1007
#18 11.43 Progress: resolved 1104, reused 0, downloaded 1099, added 1098
#18 12.43 Progress: resolved 1104, reused 0, downloaded 1101, added 1101
#18 13.93 Progress: resolved 1104, reused 0, downloaded 1102, added 1101
#18 15.48 Progress: resolved 1104, reused 0, downloaded 1102, added 1102
#18 16.67 Progress: resolved 1104, reused 0, downloaded 1103, added 1102
#18 17.67 Progress: resolved 1104, reused 0, downloaded 1103, added 1103
#18 18.73 Progress: resolved 1104, reused 0, downloaded 1104, added 1103
#18 19.53 Progress: resolved 1104, reused 0, downloaded 1104, added 1104, done
#18 19.85 .../node_modules/@prisma/engines postinstall$ node scripts/postinstall.js
#18 19.85 .../node_modules/@tailwindcss/oxide postinstall$ node ./scripts/install.js
#18 19.91 .../argon2@0.43.0/node_modules/argon2 install$ node-gyp-build
#18 19.91 .../sharp@0.34.2/node_modules/sharp install$ node install/check
#18 19.91 .../node_modules/@sentry/cli postinstall$ node ./scripts/install.js
#18 20.02 .../node_modules/@tailwindcss/oxide postinstall: Done
#18 20.03 .../node_modules/@sentry/cli postinstall: Done
#18 20.08 .../argon2@0.43.0/node_modules/argon2 install: Done
#18 20.10 .../sharp@0.34.2/node_modules/sharp install: Done
#18 21.47 .../node_modules/@prisma/engines postinstall: Done
#18 21.59 .../node_modules/prisma preinstall$ node scripts/preinstall-entry.js
#18 21.67 .../node_modules/prisma preinstall: Done
#18 21.77 .../node_modules/@prisma/client postinstall$ node scripts/postinstall.js
#18 23.87 .../node_modules/@prisma/client postinstall: Prisma schema loaded from prisma/schema.prisma
#18 24.08 .../node_modules/@prisma/client postinstall: ✔ Generated Prisma Client (v6.10.1) to ./node_modules/.pnpm/@prisma+client@6.10.1_prisma@6.10.1_typescript@5.8.3__typescript@5.8.3/node_modules/@prisma/client in 60ms
#18 24.08 .../node_modules/@prisma/client postinstall: Start by importing your Prisma Client (See: https://pris.ly/d/importing-client)
#18 24.08 .../node_modules/@prisma/client postinstall: Tip: Want to turn off tips and other hints? https://pris.ly/tip-4-nohints
#18 24.08 .../node_modules/@prisma/client postinstall: ┌─────────────────────────────────────────────────────────┐
#18 24.08 .../node_modules/@prisma/client postinstall: │  Update available 6.10.1 -> 6.11.1                      │
#18 24.08 .../node_modules/@prisma/client postinstall: │  Run the following to update                            │
#18 24.08 .../node_modules/@prisma/client postinstall: │    npm i --save-dev prisma@latest                       │
#18 24.08 .../node_modules/@prisma/client postinstall: │    npm i @prisma/client@latest                          │
#18 24.08 .../node_modules/@prisma/client postinstall: └─────────────────────────────────────────────────────────┘
#18 24.10 .../node_modules/@prisma/client postinstall: Done
#18 24.36
#18 24.36 dependencies:
#18 24.36 + @aws-sdk/client-s3 3.842.0
#18 24.36 + @dnd-kit/core 6.3.1
#18 24.36 + @dnd-kit/sortable 10.0.0
#18 24.36 + @dnd-kit/utilities 3.2.2
#18 24.36 + @fortawesome/fontawesome-svg-core 6.7.2
#18 24.36 + @fortawesome/free-solid-svg-icons 6.7.2
#18 24.36 + @fortawesome/react-fontawesome 0.2.2
#18 24.36 + @hookform/resolvers 5.1.1
#18 24.36 + @next/third-parties 15.4.0-canary.51
#18 24.36 + @prisma/client 6.10.1
#18 24.36 + @radix-ui/react-checkbox 1.3.2
#18 24.36 + @radix-ui/react-dialog 1.1.14
#18 24.36 + @radix-ui/react-dropdown-menu 2.1.15
#18 24.36 + @radix-ui/react-label 2.1.7
#18 24.36 + @radix-ui/react-popover 1.1.14
#18 24.36 + @radix-ui/react-progress 1.1.7
#18 24.36 + @radix-ui/react-radio-group 1.3.7
#18 24.36 + @radix-ui/react-scroll-area 1.2.9
#18 24.36 + @radix-ui/react-select 2.2.5
#18 24.36 + @radix-ui/react-separator 1.1.7
#18 24.36 + @radix-ui/react-slot 1.2.3
#18 24.36 + @radix-ui/react-switch 1.2.5
#18 24.36 + @radix-ui/react-tooltip 1.2.7
#18 24.36 + @sentry/nextjs 9.33.0
#18 24.36 + @tailwindcss/postcss 4.1.11
#18 24.36 + @tailwindcss/typography 0.5.16
#18 24.36 + @types/d3 7.4.3
#18 24.36 + @types/disposable-email 0.2.2
#18 24.36 + @types/nodemailer 6.4.17
#18 24.36 + @types/ws 8.18.1
#18 24.36 + argon2 0.43.0
#18 24.36 + babel-plugin-react-compiler 19.1.0-rc.2
#18 24.36 + class-variance-authority 0.7.1
#18 24.36 + clsx 2.1.1
#18 24.36 + cmdk 1.1.1
#18 24.36 + d3 7.9.0
#18 24.36 + disposable-email 0.2.3
#18 24.36 + google-translate-api-x 10.7.2
#18 24.36 + import-in-the-middle 1.14.2
#18 24.36 + jdenticon 3.3.0
#18 24.36 + jose 6.0.11
#18 24.36 + lottie-react 2.4.1
#18 24.36 + lucide-react 0.524.0
#18 24.36 + motion 12.19.2
#18 24.36 + next 15.4.0-canary.23
#18 24.36 + next-pwa 5.6.0
#18 24.36 + next-view-transitions 0.3.4
#18 24.36 + next-ws 2.0.9
#18 24.36 + nextstepjs 2.1.1
#18 24.36 + nodemailer 7.0.3
#18 24.36 + react 19.1.0
#18 24.36 + react-dom 19.1.0
#18 24.36 + react-hook-form 7.59.0
#18 24.36 + react-infinite-scroll-component 6.1.0
#18 24.36 + react-loader-spinner 6.1.6
#18 24.36 + react-markdown 10.1.0
#18 24.36 + react-scroll-motion 0.3.5
#18 24.36 + react-toastify 11.0.5
#18 24.36 + rehype-highlight 7.0.2
#18 24.36 + rehype-sanitize 6.0.0
#18 24.36 + remark-gfm 4.0.1
#18 24.36 + require-in-the-middle 7.5.2
#18 24.36 + sharp 0.34.2
#18 24.36 + tailwind-merge 3.3.1
#18 24.36 + tailwindcss-animate 1.0.7
#18 24.36 + uuid 11.1.0
#18 24.36 + ws 8.18.3
#18 24.36 + zod 3.25.67
#18 24.36
#18 24.36 devDependencies:
#18 24.36 + @eslint/js 9.30.0
#18 24.36 + @types/node 24.0.7
#18 24.36 + @types/react 19.1.8
#18 24.36 + @types/react-dom 19.1.6
#18 24.36 + eslint 9.30.0
#18 24.36 + eslint-config-prettier 10.1.5
#18 24.36 + eslint-plugin-import 2.32.0
#18 24.36 + eslint-plugin-react 7.37.5
#18 24.36 + globals 16.2.0
#18 24.36 + postcss 8.5.6
#18 24.36 + prisma 6.10.1
#18 24.36 + tailwindcss 4.1.11
#18 24.36 + typescript 5.8.3
#18 24.36 + typescript-eslint 8.35.0
#18 24.36
#18 24.40
#18 24.40 > polarlearn@v1-PRERELEASE-1.1 postinstall /app
#18 24.40 > prisma db push
#18 24.40
#18 25.21 Prisma schema loaded from prisma/schema.prisma
#18 25.21 Datasource "db": MongoDB database "polarlearn" at "192.168.86.105:27017"
#18 25.24 Error: P1013
#18 25.24
#18 25.24 The provided database string is invalid. An invalid argument was provided: connection string options is not a `key=value` pair: directConnectiontrue in database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.
#18 25.24
#18 25.28  ELIFECYCLE  Command failed with exit code 1.
#18 ERROR: process "pnpm install --frozen-lockfile --prefer-offline" did not complete successfully: exit code: 1
------
> pnpm install --frozen-lockfile --prefer-offline:
24.40 > polarlearn@v1-PRERELEASE-1.1 postinstall /app
24.40 > prisma db push
24.40
25.21 Prisma schema loaded from prisma/schema.prisma
25.21 Datasource "db": MongoDB database "polarlearn" at "192.168.86.105:27017"
25.24 Error: P1013
25.24
25.24 The provided database string is invalid. An invalid argument was provided: connection string options is not a `key=value` pair: directConnectiontrue in database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.
25.24
25.28  ELIFECYCLE  Command failed with exit code 1.
------
ERROR: failed to build: failed to solve: process "pnpm install --frozen-lockfile --prefer-offline" did not complete successfully: exit code: 1
❌ Railpack build failed
===================================EXTRA LOGS============================================
Error occurred ❌, check the logs for details.
Command exited with code 1. Stderr: ERROR: existing instance for "builder-containerd" but no append mode, specify the node name to make changes for existing instances
, command: set -e;
rm -rf /etc/dokploy/applications/polarlearn-polarlearn-y41mka/code;
mkdir -p /etc/dokploy/applications/polarlearn-polarlearn-y41mka/code;
if ! git clone --branch stable --depth 1  --progress https://oauth2:ghs_R1CpMuQ2xtQxexnqUt2ON8vh6K2rLo10cxES@github.com/polarnl/PolarLearn.git /etc/dokploy/applications/polarlearn-polarlearn-y41mka/code >> /etc/dokploy/logs/polarlearn-polarlearn-y41mka/polarlearn-polarlearn-y41mka-2025-07-05:18:58:11.log 2>&1; then
echo "❌ [ERROR] Fail to clone repository github.com/polarnl/PolarLearn.git" >> /etc/dokploy/logs/polarlearn-polarlearn-y41mka/polarlearn-polarlearn-y41mka-2025-07-05:18:58:11.log;
exit 1;
fi
echo "Cloned github.com/polarnl/PolarLearn.git to /etc/dokploy/applications/polarlearn-polarlearn-y41mka/code: ✅" >> /etc/dokploy/logs/polarlearn-polarlearn-y41mka/polarlearn-polarlearn-y41mka-2025-07-05:18:58:11.log;
# Ensure we have a builder with containerd
docker buildx create --use --name builder-containerd --driver docker-container || true
docker buildx use builder-containerd
echo "Preparing Railpack build plan..." >> "/etc/dokploy/logs/polarlearn-polarlearn-y41mka/polarlearn-polarlearn-y41mka-2025-07-05:18:58:11.log";
railpack prepare /etc/dokploy/applications/polarlearn-polarlearn-y41mka/code/ --plan-out /etc/dokploy/applications/polarlearn-polarlearn-y41mka/code//railpack-plan.json --info-out /etc/dokploy/applications/polarlearn-polarlearn-y41mka/code//railpack-info.json --env 'DATABASE_URL=mongodb://redacted:redacted@192.168.86.105:27017/polarlearn?directConnection=true&replicaSet=rs0&authSource=admin' --env 'GITHUB_ID=redacted' --env 'GITHUB_SECRET=redacted' --env 'GOOGLE_ID=redacted' --env 'GOOGLE_SECRET=redacted' --env 'NEXT_PUBLIC_SENTRY_DSN=redacted' --env 'NEXT_PUBLIC_TURNSTILE_SITE_KEY=redacted' --env 'NEXT_PUBLIC_URL=https://polarlearn.tech' --env 'PEPPER=redacted' --env 'S3_ACCESS_KEY_ID=redacted' --env 'S3_BUCKET=polarlearn' --env 'S3_ENDPOINT=https://cdn.polarlearn.tech' --env 'S3_SECRET_ACCESS_KEY=redacted' --env 'SECRET=redacted' --env 'SENTRY_AUTH_TOKEN=redacted' --env 'SERVERLESS=true' --env 'SMTP_HOST=redacted' --env 'SMTP_PASSWORD=redacted' --env 'SMTP_PORT=587' --env 'SMTP_USER=redacted' --env 'TURNSTILE_SECRET_KEY=redacted' >> /etc/dokploy/logs/polarlearn-polarlearn-y41mka/polarlearn-polarlearn-y41mka-2025-07-05:18:58:11.log 2>> /etc/dokploy/logs/polarlearn-polarlearn-y41mka/polarlearn-polarlearn-y41mka-2025-07-05:18:58:11.log || {
echo "❌ Railpack prepare failed" >> /etc/dokploy/logs/polarlearn-polarlearn-y41mka/polarlearn-polarlearn-y41mka-2025-07-05:18:58:11.log;
exit 1;
}
echo "✅ Railpack prepare completed." >> /etc/dokploy/logs/polarlearn-polarlearn-y41mka/polarlearn-polarlearn-y41mka-2025-07-05:18:58:11.log;
echo "Building with Railpack frontend..." >> "/etc/dokploy/logs/polarlearn-polarlearn-y41mka/polarlearn-polarlearn-y41mka-2025-07-05:18:58:11.log";
# Export environment variables for secrets
export DATABASE_URL='mongodb://redacted:redacted@192.168.86.105:27017/polarlearn?directConnectiontrue&replicaSetrs0&authSourceadmin'
export GITHUB_ID=redacted
export GITHUB_SECRET=redacted
export GOOGLE_ID=redacted
export GOOGLE_SECRET=redacted
export NEXT_PUBLIC_SENTRY_DSN=redacted
export NEXT_PUBLIC_TURNSTILE_SITE_KEY=redacted
export NEXT_PUBLIC_URL='https://polarlearn.tech'
export PEPPER=redacted
export S3_ACCESS_KEY_ID=redacted
export S3_BUCKET='polarlearn'
export S3_ENDPOINT=redacted
export S3_SECRET_ACCESS_KEY=redacted
export SECRET=redacted
export SENTRY_AUTH_TOKEN=redacted
export SERVERLESS='true'
export SMTP_HOST=redacted
export SMTP_PASSWORD=redacted
export SMTP_PORT=redacted
export SMTP_USER=redacted
export TURNSTILE_SECRET_KEY=redacted
docker buildx build --build-arg secrets-hash=11c94a08b2a4745c9784145ec7a7d87afa877a38da1b84b286516a89af8d585e --build-arg cache-key=_Ib6jTWbsL --build-arg BUILDKIT_SYNTAX=ghcr.io/railwayapp/railpack-frontend:v0.0.64 -f /etc/dokploy/applications/polarlearn-polarlearn-y41mka/code//railpack-plan.json --output type=docker,name=polarlearn-polarlearn-y41mka --secret id=DATABASE_URL,env=DATABASE_URL --secret id=GITHUB_ID,env=GITHUB_ID --secret id=GITHUB_SECRET,env=GITHUB_SECRET --secret id=GOOGLE_ID,env=GOOGLE_ID --secret id=GOOGLE_SECRET,env=GOOGLE_SECRET --secret id=NEXT_PUBLIC_SENTRY_DSN,env=NEXT_PUBLIC_SENTRY_DSN --secret id=NEXT_PUBLIC_TURNSTILE_SITE_KEY,env=NEXT_PUBLIC_TURNSTILE_SITE_KEY --secret id=NEXT_PUBLIC_URL,env=NEXT_PUBLIC_URL --secret id=PEPPER,env=PEPPER --secret id=S3_ACCESS_KEY_ID,env=S3_ACCESS_KEY_ID --secret id=S3_BUCKET,env=S3_BUCKET --secret id=S3_ENDPOINT,env=S3_ENDPOINT --secret id=S3_SECRET_ACCESS_KEY,env=S3_SECRET_ACCESS_KEY --secret id=SECRET,env=SECRET --secret id=SENTRY_AUTH_TOKEN,env=SENTRY_AUTH_TOKEN --secret id=SERVERLESS,env=SERVERLESS --secret id=SMTP_HOST,env=SMTP_HOST --secret id=SMTP_PASSWORD,env=SMTP_PASSWORD --secret id=SMTP_PORT,env=SMTP_PORT --secret id=SMTP_USER,env=SMTP_USER --secret id=TURNSTILE_SECRET_KEY,env=TURNSTILE_SECRET_KEY /etc/dokploy/applications/polarlearn-polarlearn-y41mka/code/ >> /etc/dokploy/logs/polarlearn-polarlearn-y41mka/polarlearn-polarlearn-y41mka-2025-07-05:18:58:11.log 2>> /etc/dokploy/logs/polarlearn-polarlearn-y41mka/polarlearn-polarlearn-y41mka-2025-07-05:18:58:11.log || {
echo "❌ Railpack build failed" >> /etc/dokploy/logs/polarlearn-polarlearn-y41mka/polarlearn-polarlearn-y41mka-2025-07-05:18:58:11.log;
exit 1;
}
echo "✅ Railpack build completed." >> /etc/dokploy/logs/polarlearn-polarlearn-y41mka/polarlearn-polarlearn-y41mka-2025-07-05:18:58:11.log;

--- a/packages/server/src/utils/docker/utils.ts
+++ b/packages/server/src/utils/docker/utils.ts
@@ -370,53 +370,53 @@ export const generateConfigContainer = (application: ApplicationNested) => {
 		}),
 		...(restartPolicySwarm
 			? {
-				RestartPolicy: restartPolicySwarm,
-			}
+					RestartPolicy: restartPolicySwarm,
+				}
 			: {}),
 		...(placementSwarm
 			? {
-				Placement: placementSwarm,
-			}
+					Placement: placementSwarm,
+				}
 			: {
-				// if app have mounts keep manager as constraint
-				Placement: {
-					Constraints: haveMounts ? ["node.role==manager"] : [],
-				},
-			}),
+					// if app have mounts keep manager as constraint
+					Placement: {
+						Constraints: haveMounts ? ["node.role==manager"] : [],
+					},
+				}),
 		...(labelsSwarm && {
 			Labels: labelsSwarm,
 		}),
 		...(modeSwarm
 			? {
-				Mode: modeSwarm,
-			}
+					Mode: modeSwarm,
+				}
 			: {
-				// use replicas value if no modeSwarm provided
-				Mode: {
-					Replicated: {
-						Replicas: replicas,
+					// use replicas value if no modeSwarm provided
+					Mode: {
+						Replicated: {
+							Replicas: replicas,
+						},
 					},
-				},
-			}),
+				}),
 		...(rollbackConfigSwarm && {
 			RollbackConfig: rollbackConfigSwarm,
 		}),
 		...(updateConfigSwarm
 			? { UpdateConfig: updateConfigSwarm }
 			: {
-				// default config if no updateConfigSwarm provided
-				UpdateConfig: {
-					Parallelism: 1,
-					Order: "start-first",
-				},
-			}),
+					// default config if no updateConfigSwarm provided
+					UpdateConfig: {
+						Parallelism: 1,
+						Order: "start-first",
+					},
+				}),
 		...(networkSwarm
 			? {
-				Networks: networkSwarm,
-			}
+					Networks: networkSwarm,
+				}
 			: {
-				Networks: [{ Target: "dokploy-network" }],
-			}),
+					Networks: [{ Target: "dokploy-network" }],
+				}),
 	};
 };
 

--- a/packages/server/src/utils/docker/utils.ts
+++ b/packages/server/src/utils/docker/utils.ts
@@ -287,7 +287,7 @@ export const parseEnvironmentKeyValuePair = (
 		throw new Error(`Invalid environment variable pair: ${pair}`);
 	}
 
-	return [key, valueParts.join("")];
+	return [key, valueParts.join("=")];
 };
 
 export const getEnviromentVariablesObject = (
@@ -370,53 +370,53 @@ export const generateConfigContainer = (application: ApplicationNested) => {
 		}),
 		...(restartPolicySwarm
 			? {
-					RestartPolicy: restartPolicySwarm,
-				}
+				RestartPolicy: restartPolicySwarm,
+			}
 			: {}),
 		...(placementSwarm
 			? {
-					Placement: placementSwarm,
-				}
+				Placement: placementSwarm,
+			}
 			: {
-					// if app have mounts keep manager as constraint
-					Placement: {
-						Constraints: haveMounts ? ["node.role==manager"] : [],
-					},
-				}),
+				// if app have mounts keep manager as constraint
+				Placement: {
+					Constraints: haveMounts ? ["node.role==manager"] : [],
+				},
+			}),
 		...(labelsSwarm && {
 			Labels: labelsSwarm,
 		}),
 		...(modeSwarm
 			? {
-					Mode: modeSwarm,
-				}
+				Mode: modeSwarm,
+			}
 			: {
-					// use replicas value if no modeSwarm provided
-					Mode: {
-						Replicated: {
-							Replicas: replicas,
-						},
+				// use replicas value if no modeSwarm provided
+				Mode: {
+					Replicated: {
+						Replicas: replicas,
 					},
-				}),
+				},
+			}),
 		...(rollbackConfigSwarm && {
 			RollbackConfig: rollbackConfigSwarm,
 		}),
 		...(updateConfigSwarm
 			? { UpdateConfig: updateConfigSwarm }
 			: {
-					// default config if no updateConfigSwarm provided
-					UpdateConfig: {
-						Parallelism: 1,
-						Order: "start-first",
-					},
-				}),
+				// default config if no updateConfigSwarm provided
+				UpdateConfig: {
+					Parallelism: 1,
+					Order: "start-first",
+				},
+			}),
 		...(networkSwarm
 			? {
-					Networks: networkSwarm,
-				}
+				Networks: networkSwarm,
+			}
 			: {
-					Networks: [{ Target: "dokploy-network" }],
-				}),
+				Networks: [{ Target: "dokploy-network" }],
+			}),
 	};
 };
 

